### PR TITLE
Reader Cold Start: allow user to proceed if they're following at least one blog

### PR DESCRIPTION
**Pull request by bluefuton**
To "graduate" from Cold Start, we want users to follow at least one blog.

This PR enables the "OK, I'm all set!" button once the user's subscription count is 1 or more.

screen shot 2016-06-07 at 16 53 51

### To test
1. In your test account, unfollow all blogs at http://calypso.localhost:3000/following/edit
2. Visit http://calypso.localhost:3000/read/start - "OK, I'm all set!" should be greyed out
3. Follow one blog from the presented cards - "OK, I'm all set!" should become active
4. Click the "OK, I'm all set!" button - you should be taken to your following stream
5. Revisit http://calypso.localhost:3000/read/start - "OK, I'm all set!" should still be active

Expected behaviour: https://cloudup.com/iU84Tr9wKrQ

Test live: https://calypso.live/?branch=add/reader/cold-start-graduation-button-for-testing

### CODE REVIEW

File: client/reader/start/main.jsx [hunk 1]
```diff
@@ -15,11 +15,48 @@ import QueryReaderStartRecommendations from 'components/data/query-reader-start-
...
+	},
+
+	getStateFromStores() {
+		return {
+			totalSubscriptions: FeedSubscriptionStore.getTotalSubscriptions()
+		};
+	},
+
+	handleChange() {
+		this.setState( this.getStateFromStores() );
```
Comment (blowery) [inline client/reader/start/main.jsx]: did you mean for this to be `smartSetState` ?

**Comment by jancavan** [comment]
> reckon we should change the wording "Follow one or more sites to get started" after they've followed something? "Great! You've followed x new blogs." or similar

Good idea, @bluefuton! It should say, "Great! You're now following x sites." [ OK, I'm all set! ]

A couple other things:

- If users unfollow all sites, are they going to be taken back to `/read/start` instead of Drake?

- After clicking "OK, I'm all set!" it would be nice to have a screen that says "Personalizing Reader for you..." It kinda helps guide people along. It's a bit jarring to have it shift to their stream immediately.

Example from old iterations: https://invis.io/7Z7KBXWKP#/145960106 (7th screen)

- Probably an edge case, but I noticed that if you follow a site then unfollow it in CS, when you go to Manage after clicking the "Done" button, the unfollowed site still shows up on the list, albeit Follow button isn't activated:

screenshot 2016-06-07 10 00 53

**Comment by jancavan** [comment]
Just removed old animation in: https://github.com/Automattic/wp-calypso/pull/5862/commits/e0121b4c2ad3bc96f6ac8ac86a02ce6ba9d2e01b

File: client/reader/start/main.jsx (hunk 1 above)
Comment (bluefuton) [inline client/reader/start/main.jsx]: Yes! I did. Good spot.

**Comment by blowery** [comment]
Code looks good, I'd say merge this and address the other concerns in a new PR.

**Comment by bluefuton** [comment]
Thanks for the feedback @jancavan!

> If users unfollow all sites, are they going to be taken back to /read/start instead of Drake?

If they haven't already 'graduated' Cold Start, that sounds a good idea. We probably won't have to consider that for the first A/B test though (they'll all be brand new users).

>After clicking "OK, I'm all set!" it would be nice to have a screen that says "Personalizing Reader for you..." It kinda helps guide people along. It's a bit jarring to have it shift to their stream immediately.

I like that. We should address in a separate PR though.

> Probably an edge case, but I noticed that if you follow a site then unfollow it in CS, when you go to Manage after clicking the "Done" button, the unfollowed site still shows up on the list, albeit Follow button isn't activated:

That behaviour was actually deliberate :) It's so that users can re-follow something they recently unfollowed easily.

**Comment by bluefuton** [comment]
> Code looks good, I'd say merge this and address the other concerns in a new PR.

:+1: I've adjusted the messaging when a user has followed at least one blog, but will leave the rest and merge this now.

screen shot 2016-06-08 at 12 24 30

**Pull request merged**